vllm/model_executor/models/transformers.py

```diff
@@ -72,15 +74,64 @@ def vllm_flash_attention_forward(
 ALL_ATTENTION_FUNCTIONS["vllm"] = vllm_flash_attention_forward
 
 
+def replace_rms_norm_class(rms_norm: nn.Module) -> RMSNorm:
+    """
+    Replace Transformers RMS norm class with vLLM's RMSNorm class.
+
+    Args:
+        rms_norm (nn.Module): RMS norm module to be replaced.
+
+    Returns:
+        RMSNorm: The new RMSNorm.
+    """
+    # Get hidden size
+    parameters = dict(rms_norm.named_parameters())
+    if len(parameters) != 1:
+        class_name = rms_norm.__class__.__name__
+        logger.warning(
+            "Unable to determine `hidden_size` of %s. "
+            "This layer will not benefit from vLLM's custom ops.", class_name)
+        return rms_norm
+    weight = next(iter(parameters.values()))
+    if weight is None and isinstance(rms_norm, nn.RMSNorm):
+        hidden_size = prod(rms_norm.normalized_shape)
+    else:
+        hidden_size = weight.numel()
+
+    # Get eps
+    attrs = vars(rms_norm)
+    condition = lambda k, v: not k.startswith("_") and isinstance(v, float)
+    attrs = {k: v for k, v in attrs.items() if condition(k, v)}
+    if len(attrs) != 1:
+        class_name = rms_norm.__class__.__name__
+        logger.warning(
+            "Unable to determine `eps` of %s. "
+            "This layer will not benefit from vLLM's custom ops.", class_name)
+        return rms_norm
+    eps = next(iter(attrs.values()))
```

Comment:
why don't we take the eps from the config?

Reply:
From a quick scan of `src/transformers/models` it appears that, similarly to `hidden_size`, the same config key is not always used for `eps`.

i.e. `modeling_falcon_mamba.py` which uses `config.layer_norm_epsilon`.